src/composables/useFlawAffectsModel.ts

```diff
@@ -123,9 +124,14 @@ export function useFlawAffectsModel(flaw: Ref<ZodFlawType>) {
     theAffects.value.push(recoveredAffect);
   }
 
+  function hasAffectChanged(affect: ZodAffectType) {
+    const originalAffect = initialAffects.find((maybeMatch) => maybeMatch.uuid === affect.uuid);
+    return !equals(originalAffect, affect);
+  }
+
   theAffects.value.forEach((affect) => {
     watch(affect, () => {
-      if (affect.uuid) {
+      if (affect.uuid && hasAffectChanged(affect)) {
```

Comment:
[watch](https://vuejs.org/guide/essentials/watchers.html#basic-example) callbacks have the current and previous values as parameters, is that not enough? Why do we need to use the `initialAffects` array? Just curious

```js
 watch(affect, (newAffect, oldAffect) => {
```
```suggestion
      if (affect.uuid && !equals(newAffect, oldAffect)) {
```
> github does not allow me to suggest changes on line 133 for some reason

Reply:
That's a great question! It frustratingly does not work that way with properties of reactives :(
From https://vuejs.org/guide/essentials/watchers.html#deep-watchers
```js
const obj = reactive({ count: 0 })

watch(obj, (newValue, oldValue) => {
  // fires on nested property mutations
  // Note: `newValue` will be equal to `oldValue` here
  // because they both point to the same object!
})

obj.count++
```

Reply:
Fair point there, but if we only need to use the `uuid` of the affect, shouldn't it be easier just to watch the uuid?
That way the `{ deep: true }` can be removed
```js
theAffects.value.forEach((affect) => {
  watch(()=>affect.uuid, () => {
    reportAffectAsModified(uuid);
  })
})